draft-ietf-quic-transport.md

```diff
@@ -1915,23 +1916,46 @@ than it has sent, unless this is a result of a change in the initial limits (see
 {{zerortt-parameters}}).
 
 
-## PING frame {#frame-ping}
+## PING Frame {#frame-ping}
 
 Endpoints can use PING frames (type=0x07) to verify that their peers are still
-alive or to check reachability to the peer. The PING frame contains no
-additional fields. The receiver of a PING frame simply needs to acknowledge the
-packet containing this frame.
+alive or to check reachability to the peer.
 
-A PING frame has no additional fields.
+The PING frame contains a variable-length payload.
 
-The PING frame can be used to keep a connection alive when an application or
-application protocol wishes to prevent the connection from timing out.  An
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|   Length(8)   |                 Data (*)                    ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+Length:
+
+: This 8-bit value describes the length of the Data field.
+
+Data:
+
+: This variable-length field contains arbitrary data.
```

Comment:
"Upon loss detection, a retransmitted PING MUST carry the same data."

Reply:
I don't think that's necessary.  On the contrary, it could increase the state commitment for the sender.  What is being tested is the need for a PING/PONG.  We're not trying to send data.

For reference, ICE sends similar probes and every single one of them contains a different transaction ID (the field that is analogous to the field that this PR adds), especially when the probe is being sent multiple times over a long period of time.  QUIC doesn't have the same identification problem - it can use ACKs to correlate packets - but it does have to verify that a particular was correctly received, and it can be more certain of that if the PING contains fresh entropy every time.

Reply:
Hm -- yeah, I forgot that there are ACKs for PINGs.